### PR TITLE
Kill homebrewed (v)snprintf implementations, as they are already available in the C99 standard

### PR DESCRIPTION
**Pull request by gahr**
@neomutt/reviewers Please review

### CODE REVIEW

**Comment by toogley** [comment]
@gahr what exactly means "as they are C99" ? i thought we would write C99 in mutt?

**Comment by gahr** [comment]
@toogley yes, that's the point. Those functions are in standard C99, so if we assume a C99 compiler, there's no reason to keep an implementation in case they are not there.

**Comment by d-k-c** [comment]
Amen to that. Do you know if that implies Neomutt will break on some platforms because of incomplete C99 support by some compilers?

**Comment by gahr** [comment]
@d-k-c I have no idea, sorry.

**Comment by toogley** [comment]
@neomutt/downstream-packagers do you guys know sth about [the question](https://github.com/neomutt/neomutt/pull/402#issuecomment-280063606) of @d-k-c ?

**Comment by 0-wiz-0** [comment]
That commit is no problem for NetBSD. NetBSD's man page says that both functions appeared in 4.4BSD, so it shouldn't be a problem for any of the BSDs.

**Comment by lbschenkel** [comment]
I am not an expert, but I believe that all OS X versions supported by MacPorts implement these calls.

File: mutt.h [hunk 1]
```diff
@@ -26,9 +26,6 @@
 #ifdef HAVE_UNISTD_H
 # include <unistd.h> /* needed for SEEK_SET */
 #endif
-#ifdef HAVE_UNIX_H
-# include <unix.h>   /* needed for snprintf on QNX. */
```
Comment (darnir) [inline mutt.h]: I'm not entirely sure, but if this allows us to keep compatibility with some esoteric platform, then it should be okay. 3 lines of code here won't hurt us much.

Does anyone know if Neomutt still compiles on QNX and if this ifdef is required on it?

**Comment by neverpanic** [comment]
Even PPC Macs had those functions already, so this is fine from a Mac perspective.

File: mutt.h (hunk 1 above)
Comment (guyzmo) [inline mutt.h]: I searched through the docs of QNX, and found that ① they're C99 compliant and ② there's a [`stdio.h` featuring snprintf](http://www.qnx.com/developers/docs/6.5.0SP1.update/index.html#./com.qnx.doc.dinkum_en_c99/stdio.html).

So I vote for removing those useless three lines.

**Comment by xdgc** [comment]
* On 15 Feb 2017, Damien R. wrote: 
> Amen to that. Do you know if that implies Neomutt will break on some platforms because of incomplete C99 support by some compilers?

It certainly will.  Some compilers were made before '99....

-- 
David Champion • dgc@bikeshed.us

**Comment by Vaelatern** [comment]
It would break support for old compilers, but remotely modern gcc and friends have c99 available.

**Comment by d-k-c** [comment]
> It certainly will. Some compilers were made before '99....

Mutt requires C99 since August 2015, so the question is "are there compilers that claim to be C99 compliant but which actually aren't".

**Comment by guyzmo** [comment]
> the question is "are there compilers that claim to be C99 compliant but which actually aren't".

and the answer should be *screw them*.

If you're stuck with a system that does not have a C99 compliant compiler on which you cannot build one (like GCC) then you're stuck with using an old version of mutt.

**Comment by toogley** [comment]
@d-k-c 
> Mutt requires C99 since August 2015, so the question is "are there compilers that claim to be C99 compliant but which actually aren't".

What's the source for that? I mean, can you cite your source?^^

**Comment by flatcap** [comment]
@toogley 
`AC_PROG_CC_C99` was added to `configure.ac` by Kevin McCarthy on 2015-08-17 (a9676b02)
This means a C99-compliant compiler is required.

**Comment by gahr** [comment]
Plus, [this check](https://github.com/neomutt/neomutt/blob/master/configure.ac#L33-L35)

**Comment by xdgc** [comment]
* On 16 Feb 2017, guyzmo wrote: 
> > the question is "are there compilers that claim to be C99 compliant but which actually aren't".
> 
> and the answer should be *screw them*.
> 
> If you're stuck with a system that does not have a C99 compliant compiler on which you cannot build one (like GCC) then you're stuck with using an old version of mutt.

I talked about that in another thread.  Neomutt can do what you like of
course, but mutt should not abandon users without making it clear in
advance that there's a cutoff coming.


> @d-k-c 
> > Mutt requires C99 since August 2015, so the question is "are there compilers that claim to be C99 compliant but which actually aren't".

I didn't remember this.  If that decision has been made already, then I
have no objection on the basis of a support obligation.

But once again, I'm not sure what's the big win in removing passive
compatibility code that hasn't needed maintenance in over ten years.
The only thing achieved is less user support.  Nothing future-looking is
enabled by the change.

-- 
David Champion • dgc@bikeshed.us

**Comment by guyzmo** [comment]
On Thu, Feb 16, 2017 at 02:29:48PM -0800, David Champion wrote:
> I talked about that in another thread.  Neomutt can do what you like of
> course, but mutt should not abandon users without making it clear in
> advance that there's a cutoff coming.

Of course, my former remark is based on the fact below. ↓↓↓

> > @d-k-c 
> > > Mutt requires C99 since August 2015 […]
> I didn't remember this.  If that decision has been made already, then I
> have no objection on the basis of a support obligation.

👍

> But once again, I'm not sure what's the big win in removing passive
> compatibility code that hasn't needed maintenance in over ten years.

It's about removing bitrot, the part of code that, in theory, will never
get included at compile time by any user. And somehow, the only way to
prove that theory, it's to remove if from the codebase and wait for
someone to yell about it.

> The only thing achieved is less user support.  Nothing future-looking is
> enabled by the change.

It's like taking the garbage out for collecting. Nothing future-looking
is enabled by taking the garbage out, but you're still taking them out! ☺

-- 
Guyzmo

**Pull request merged**